src/NServiceBus.Core/Recoverability/RecoverabilityAction.cs

```diff
@@ -58,13 +58,23 @@ public static Discard Discard(string reason)
             Guard.AgainstNullAndEmpty(nameof(reason), reason);
             return new Discard(reason);
         }
-
+        
         /// <summary>
         /// Creates a custom recoverability action.
         /// </summary>
         /// <param name="customAction">The action to perform for the failed message.</param>
         /// <returns>Custom recoverability action.</returns>
         public static CustomAction Custom(Func<ErrorContext, Task> customAction)
+        {
+            return new CustomAction((e, d) => customAction(e));
+        }        
+
+        /// <summary>
+        /// Creates a custom recoverability action.
+        /// </summary>
+        /// <param name="customAction">The action to perform for the failed message.</param>
+        /// <returns>Custom recoverability action.</returns>
+        public static CustomAction Custom(Func<ErrorContext, IDispatchMessages, Task> customAction)
```

Comment:
wondering whether we should just provide this overload given it's a very advanced API anyway?

Reply:
Yeah maybe less is more. I'll let you drive this